### ISSUE TITLE
Is there any source code for the Demo?

### ISSUE THREAD

**Issue by jeffersonaraujo** [marklcg/jsf-sheet#5]
I'm trying to fire an ajax event in my manage bean but I do not know what kind of event is accepted, I did not see any example. Can you help me?

**Comment by marklcg** [comment]
It has been on my list to make the source for the demo available for a while.  It is long overdue.  I will work on that, but in the meantime, here is an example of the ajax event from the demo:

```xml
<p:ajax listener="#{assetController.cellEditEvent}"/>
```

```java
	public void cellEditEvent(AjaxBehaviorEvent event) {
		Sheet sheet = (Sheet) event.getComponent();
		List<SheetUpdate> updates = sheet.getUpdates();
		// A SheetUpdate exists for each column updated, the same row may
		// appear more than once
		Set<Asset> processed = new HashSet<>();
		int rowUpdates = 0;
		for (SheetUpdate update : updates) {
			Asset asset = (Asset) update.getRowData();
			if (processed.contains(asset))
				continue;
			manager.update(asset);
			updateEvent.fire(asset);
		}
		// sheet.commitUpdates();
		FacesContext.getCurrentInstance().addMessage(null,
				new FacesMessage("Update Success", Integer.toString(rowUpdates) + " rows updated"));
	}
```

**Comment by jeffersonaraujo** [comment]
Thank you very much. I ended up doing some tests and found out that it was that way. If it is not bothering, I would like to know how to refresh the sheet via javascript, when I am using the JSF update the table is fading, and I believe it would need to load again with the validations.

Once again, thank you very much.


```
<p:messages id="financialSheetMessages"/>
<lcg:sheet id="financialSheet" value="#{workplanAddBean.contributions}" var="row" height="200" label="Aporte" rowKey="#{row.index}" fixedCols="1" showRowHeaders="false" stretchH="none">
									               
<p:ajax listener="#{workplanAddBean.financialSheetEditEvent}" immediate="true" update="financialSheet financialSheetMessages"/>
										        
<lcg:column headerText=" - " value="#{row.name}" colWidth="100" readonly="true"/>
										        
<c:forEach items="#{workplanAddBean.referencePeriods}" var="referencePeriod" varStatus="loop_referencePeriod">
<lcg:column id="col_#{loop_referencePeriod.index}" headerText="#{referencePeriod.description}" colType="numeric" valueChangeListener="#{workplanAddBean.financialSheetEditEvent('Teste')}" value="#{workplanAddBean.getAmountByMonthAndType(row.index == 0 ? 'CONTRIBUTION' : (row.index == 1 ? 'OUTLOOK' : 'BUDGETED'), referencePeriod) eq null ? '' : workplanAddBean.getAmountByMonthAndType(row.index == 0 ? 'CONTRIBUTION' : (row.index == 1 ? 'OUTLOOK' : 'BUDGETED'), referencePeriod)}" colWidth="110" readonlyCell="#{row.index == 0}">
<f:convertNumber pattern="#,##0.00"/>
</lcg:column>
</c:forEach>
</lcg:sheet>
```

The problem is in this update: update="financialSheet financialSheetMessages", specifically here: financialSheet

Here's a print of how it looks
<img width="1033" alt="screen shot 2016-12-05 at 05 16 38" src="https://cloud.githubusercontent.com/assets/607226/20877881/4ece04e6-baaa-11e6-94e1-6abaf34fab2f.png">

**Comment by marklcg** [comment]
I'm not sure why it's doing that, but it should update validations on it's (leaving off the sheet id).  Also, you can try updating the form: 

```xml
update="@form"
```

I will see if I can reproduce it though.

**Comment by amolksagar** [comment]
Hi Mark and Jefferson, 

       I am trying to use jsf-sheet 1.1.

I am not able to see the table columns because of the following errors:


**_handsontable.js?ln=handsontable:11373 **Uncaught ReferenceError: jQuery is not defined_**

**Uncaught ReferenceError: PrimeFaces is not defined**

This error comes when I launch the xhtml page.

Please Advise.

**Comment by amolksagar** [comment]
Hi All,

  I am getting the following error on executing my event Handler.Please help me with this .I am stuck because of this while using jsf-sheet

**Error**:
Uncaught ReferenceError: widget_j_idt3_sheet is not defined
    at eval (eval at <anonymous> (jquery.js?ln=primefaces&v=6.0:14), <anonymous>:1:1)
    at eval (<anonymous>)
    at jquery.js?ln=primefaces&v=6.0:14
    at Function.globalEval (jquery.js?ln=primefaces&v=6.0:14)
    at Object.doEval (core.js?ln=primefaces&v=6.0:3)
    at Object.handle (core.js?ln=primefaces&v=6.0:3)
    at Object.success (core.js?ln=primefaces&v=6.0:3)
    at i (jquery.js?ln=primefaces&v=6.0:25)
    at Object.fireWith [as resolveWith] (jquery.js?ln=primefaces&v=6.0:25)
    at cf (jquery.js?ln=primefaces&v=6.0:25)


**My Event handler**:
 public void buttonAction(ActionEvent actionEvent) {
	Sheet sheet = (Sheet)actionEvent.getComponent().getParent().getChildren().get(1);
      	FacesMessage message = new FacesMessage(FacesMessage.SEVERITY_INFO, sheet.getSelection(),  null);
    	FacesContext.getCurrentInstance().addMessage(null, message);
}


My **index.xhtml**:

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" 
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<h:html
    xmlns:p="http://primefaces.org/ui" 
	xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:ui="http://java.sun.com/jsf/facelets"
	xmlns:lcg="http://www.lassitercg.com/jsf">
<h:head>
</h:head>
<h:form>
<h:body>
 <p:commandButton value="Lock Cells" actionListener="#{dtFilterView.buttonAction}" />
   <lcg:sheet id="sheet" value="#{dtFilterView.cars}" var="car" height="400" label="Sheet" rowKey="#{car.id}">
        <f:facet name="header">
            <h:outputText value="Assets"/>
        </f:facet>

        <f:facet name="footer">
            <h:outputText value="#{dtFilterView.cars.size()} Records"/>
        </f:facet>

        <lcg:column headerText="Car Id" readonly="true" value="#{car.id}" sortBy="#{car.id}"
                    colWidth="100" filterBy="#{car.id}"/>
        <lcg:column headerText="Brand" value="#{car.brand}" sortBy="#{car.brand}"
                    colWidth="100" filterBy="#{car.brand}"/>
        <lcg:column headerText="Color" value="#{car.color}" colWidth="200" sortBy="#{car.color}"
                    filterBy="#{car.color}"/>
        <lcg:column headerText="Type" value="#{car.year}" readonly="true" colWidth="100"
                    sortBy="#{car.year}" filterBy="#{car.year}"/>
    </lcg:sheet> 
</h:body>
</h:form>
</h:html>


On clicking on the button I get the above mentioned error

**Comment by melloware** [comment]
I am sure @marklcg will respond soon with his demo project but I can tell you I am using the Sheet successfully in PF 6.0 so we have similar environments.  So hopefully it is something simple we can figure out.

**Comment by marklcg** [comment]
@amolksagar I believe the first issue you encountered was likely because you did not have any other Primefaces components on the page and the sheet component uses those libraries.  I probably need to update the Resources requirements for the component to include primefaces so those resources get included.  I thought I had done that, but will double check.

On the second issue, I will have to dig deeper.  As @melloware indicated, I will work on posting the demo code shortly.

**Comment by melloware** [comment]
@marklcg I just submitted a PR to update the dependencies to include PF.

**Comment by amolksagar** [comment]
Thanks a ton Mark and Melloware. 

Issue 1: Problem was with the order in which the libraries were loaded as Mark mentioned. For now, what I did was that I added a command button from the prime faces library before the lcg:sheet tag. That made sure that jQuery and prime faces libraries are loaded before jsf-sheet library. 

Issue 2: On the command button I added a update="@all" attribute. So once the action listener was executed the page was loaded again and the issue was resolved.
